### PR TITLE
display all aptogotchi holders to demonstrate indexer usage

### PR DESCRIPTION
**Pull request by 0xaptosj**
1. setup ci to test move code
2. add publish script
3. displaying all aptogotchi holders to demonstrate indexer usage (probably need to update the ui to make it more friendly)
4. make get aptogotchi address a view function so we can call it to get token data then collection data.
Screenshot 2023-10-18 at 6 50 19 PM

### CODE REVIEW

File: move/Move.toml [hunk 1]
```diff
@@ -4,7 +4,10 @@ version = '1.0.0'
 upgrade_policy = "compatible"
 
 [addresses]
-aptogotchi = '573503bbc3bc51ca5ff71042c548babfe604239c08c6fcd91e1a71ef44716c92'
+aptogotchi = '_'
+
+[dev-addresses]
+aptogotchi = "0x100"
```
Comment (0xaptosj) [inline move/Move.toml:10]: learned this trick from max that we can have dev address for testing so we don't need to pass it in `move test ...`

File: .github/workflows/move_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  move-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dev dependencies.
+        run: ./move/sh_scripts/dev_setup.sh
+      - name: Runs move tests.
+        run: ./move/sh_scripts/move_tests.sh
```
Comment (0xZihan) [inline .github/workflows/move_tests.yaml:18]: Noob question: what's this for?

Comment (0xZihan) [reply .github/workflows/move_tests.yaml:18]: Also what are the .idea files?

File: move/Move.toml (hunk 1 above)
Comment (0xZihan) [reply move/Move.toml:10]: can you add a comment somewhere in the code to explain it? it'll be useful for others to know

File: frontend/src/app/home/Pet/Collection.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        query MyQuery($collection_id: String) {
+          current_collection_ownership_v2_view(
+            where: { collection_id: { _eq: $collection_id } }
+          ) {
+            owner_address
+          }
+        }
+      `,
+      variables: {
+        collection_id: collectionResponse.collection_id,
```
Comment (0xZihan) [inline frontend/src/app/home/Pet/Collection.tsx:48]: can collection_id be queried from a view function in the contract? it's never gonna change so doesn't seem clean to query it every time with the token address.

File: frontend/src/app/home/Pet/Collection.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      },
+    };
+    const collectionHolderResponse = await provider.indexerClient.queryIndexer(
+      getAllCollectionHoldersGql
+    );
+
+    console.log(JSON.stringify(tokenDataResponse, null, 2));
+    console.log(
+      JSON.stringify(
+        // @ts-ignore
```
Comment (0xZihan) [inline frontend/src/app/home/Pet/Collection.tsx:58]: Is it possible to handle this properly?

**Review by 0xZihan — CHANGES_REQUESTED** [review]

File: .github/workflows/move_tests.yaml (hunk 2 above)
Comment (0xaptosj) [reply .github/workflows/move_tests.yaml:18]: .idea files are intellij config, i shuold have ignored them

Comment (0xaptosj) [reply .github/workflows/move_tests.yaml:18]: this ci first install aptos cli then run the script which runs move unit test

File: move/Move.toml (hunk 1 above)
Comment (0xaptosj) [reply move/Move.toml:11]: done

File: frontend/src/app/home/Connected.tsx [hunk 5]
```diff
@@ -10,41 +10,71 @@ export const provider = new Provider(Network.TESTNET);
...
 export function Connected() {
   const [pet, setPet] = useState<Pet>();
+  const [collectionID, setCollectionID] = useState<string>();
   const { account, network } = useWallet();
 
   const fetchPet = useCallback(async () => {
     if (!account?.address) return;
 
-    const payload = {
+    const getAptogotchiPayload = {
```
[the remark this answers is not in the document]
Comment (0xZihan) [reply frontend/src/app/home/Connected.tsx:19]: I feel it's kinda nice to give a name to it to improve readability. But maybe I missed some benefit of making it inline.

File: frontend/src/app/home/Connected.tsx [hunk 6]
```diff
@@ -10,41 +10,71 @@ export const provider = new Provider(Network.TESTNET);
...
+
+    fetchCollectionID();
+  }, [fetchCollectionID, network]);
+
   return (
     <div className="flex flex-col gap-3 p-3">
-      {pet ? <Pet pet={pet} setPet={setPet} /> : <Mint fetchPet={fetchPet} />}
+      {collectionID ? (
+        pet ? (
+          <Pet pet={pet} setPet={setPet} collectionID={collectionID} />
```
Comment (0xZihan) [inline frontend/src/app/home/Connected.tsx:73]: collectionID is not used in `Connected` or `Pet`. Let's just query collectionID in `Collection`.

**Review by 0xZihan — CHANGES_REQUESTED** [review]

File: frontend/src/app/home/Connected.tsx (hunk 6 above)
Comment (0xaptosj) [reply frontend/src/app/home/Connected.tsx]: done

File: frontend/src/utils/address.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+// In some case the leading char is 0 after the 0x and it got truncated
+// This function will add it back if needed cause indexer doesn't auto pad it
+export function padAddressIfNeeded(address: string) {
+  if (address.length === 67) {
+    return address;
+  }
+  return `0x0${address.slice(2)}`;
```
[the remark this answers is not in the document]
Comment (0xaptosj) [reply frontend/src/utils/address.ts]: updated

**Review by 0xZihan — APPROVED** [review]

**Pull request merged**